fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -873,6 +872,51 @@ public Void visitFunctionCall(FunctionCallExpr node, Scope scope) {
                     throw new SemanticException(fnName + " should have at least two inputs");
                 }
                 fn = Expr.getBuiltinFunction(fnName, argumentTypes, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+            } else if (fnName.equals("array_generate")) {
+                if (node.getChildren().size() < 1 || node.getChildren().size() > 3) {
+                    throw new SemanticException(fnName + " has wrong input numbers");
+                }
+                for (Expr expr : node.getChildren()) {
+                    if ((expr instanceof SlotRef) && node.getChildren().size() != 3) {
+                        throw new SemanticException(fnName + " with IntColumn doesn't support default parameters");
+                    }
+                    if (!(expr instanceof IntLiteral) && !(expr instanceof LargeIntLiteral) &&
+                            !(expr instanceof SlotRef) && !(expr instanceof NullLiteral)) {
+                        throw new SemanticException(fnName + "'s parameter only support Integer");
+                    }
+                }
+                // add the default parameters for array_generate
+                if (node.getChildren().size() == 1) {
+                    LiteralExpr secondParam = (LiteralExpr) node.getChild(0);
+                    node.clearChildren();
+                    try {
+                        node.addChild(new IntLiteral("1", Type.TINYINT));
```

Comment:
use secondParam().getType() instead of Type.TYNYINT, keep all of the argument types consistent.

Reply:
IntLiteral(long longValue, Type type) need catch exception, too many catch and try will make code not neat, I think just use IntLiteral(long value) is enough. CompareMode.IS_SUPERTYPE_OF will help us find the correct function anyway